lib/src/html_renderer.dart

```diff
@@ -55,13 +72,23 @@ class HtmlRenderer implements NodeVisitor {
   }
 
   void visitText(Text text) {
-    buffer.write(text.text);
+    var content = text.text;
+    if (const ['p', 'li'].contains(_lastVisitedTag)) {
```

Comment:
What a weird rule, can you link to the commonmark spec behind this?

Reply:
It's all about blocks within blocks. Updated the logic. Much better now.